### PR TITLE
feat(rolestable): delete roles modal

### PR DESCRIPTION
**Pull request by InsaneZein**
### Description
This allows user to delete Roles. Currently deleting roles api doesn't exist so in the meantime, clicking "delete role" button will just spit out a console.log message. 

[RHCLOUD-32241](https://issues.redhat.com/browse/RHCLOUD-32241)

---

### Checklist ☑️
- [ ] PR only fixes one issue or story 
- [ ] Change reviewed for extraneous code 
- [ ] UI best practices adhered to 
- [ ] Commits squashed and meaningfully named 
- [ ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

**Comment by InsaneZein** [comment]
I realized that sometimes selectedRole is undefined so the modal shows undefined instead of the role name.

File: src/smart-components/role/RolesTable.tsx [hunk 1]
```diff
@@ -118,6 +135,23 @@ const RolesTable: React.FunctionComponent<RolesTableProps> = ({ selectedRole = u
     <React.Fragment>
       <ContentHeader title="Roles" subtitle={''} />
       <PageSection isWidthLimited>
+        {isDeleteModalOpen && (
+          <WarningModal
+            ouiaId={`${ouiaId}-remove-role-modal`}
+            isOpen={isDeleteModalOpen}
```
Comment (karelhala) [inline src/smart-components/role/RolesTable.tsx:141]: Based on the mocks the modal is missing checkbox
```suggestion
            isOpen={isDeleteModalOpen}
            withCheckbox
```

**Review by karelhala — CHANGES_REQUESTED** [review]
Looks like some of the messages needs to be changed to role's messages. Also when I pulled the PR I found it's missing the toolbar actions which should allow to remove multiple roles.

**Comment by karelhala** [comment]
@InsaneZein I've checked the API and it has a delete role option https://console.redhat.com/docs/api/rbac/v1#operations-Role-deleteRole it should however be available only on custom roles. If the role is custom can be checked in role definition if `system = false`

**Review by karelhala — CHANGES_REQUESTED** [review]
The modal is using groups messages and looks like this:
Screenshot 2024-11-13 at 9 17 04

But the messages should be:
@
image-2024-09-09-12-24-18-530


The DataView is missing an action button to remove multiple roles. Also selecting single item selects the whole table.

**Comment by InsaneZein** [comment]
Ah ok the task made it sound like this was just for deleting a singular Role. I can work on adding an action button to delete multiple roles in another PR.

**Review by karelhala — APPROVED** [review]

**Pull request merged**